internal/provider/util_test.go

```diff
@@ -0,0 +1,38 @@
+package provider
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func Test_resourceName(t *testing.T) {
+	cases := []struct {
+		name                 string
+		providerShortName    string
+		templateFileName     string
+		expectedResourceName string
+	}{
+		{
```

Comment:
Can you add another case where you have a resource named (just for kicks) `https`, using the same format used by the `http` provider?

I say this because the `http` provider is here not respecting the convention: `<provider>_<resource>` when exposed.

Reply:
@detro you mean along the lines of:
```go
		{
			"provider short name different as template file name",
			"https",
			"http.md.tmpl",
			"https_http",
		},
```

Reply:
@detro I've refactored following discussion.